src/datastructure/Frame.cpp

```diff
@@ -75,6 +75,15 @@ void Frame::setKeypoints(const std::vector<Keypoint> & kpts){
     m_keypoints  = kpts;
 }
 
+bool Frame::updateKeypointClassId(int i, int classId) 
+{
+    if (i < 0 || i >= static_cast<int>(m_keypoints.size()) || i >= static_cast<int>(m_keypointsUndistort.size()))
+        return false;
+    m_keypoints[i].setClassId(classId);
+    m_keypointsUndistort[i].setClassId(classId);
```

Comment:
Does all keypoints in m_keypointsUndistort belongs to m_keypoints? I.e. is m_keypointsUndistort only keeping an information about the state of some keypoints?
If so, it could be more efficient to either have an attribute in the keypoints indicating their "undistort" state, or a vector of pointers to these keypoints (depending of what calculation we do with them).
This design decision is problably not to addressed in the PR, though.
Because here, we'd be managing copies of keypoints instances (takes memory), with the risk of missing joint updates, leading to inconsistent states (error-prone).